content/spin/v2/http-trigger.md

```diff
@@ -78,6 +78,8 @@ route = "/users/..."
 component = "user-manager"
 ```
 
+Granular routing in Spin allows users to define dynamic routes where certain parts of the URL can change based on the input provided. A colon (`:`) precedes a variable segment in a URL. This variable segment captures part of the URL as a variable, which can be used in your route handler. For example, `/goodbye/:planet`. In this route, `:planet` is a variable segment. When a user accesses a URL like `/goodbye/mars`, the router matches this to the `/goodbye/:planet` pattern, and `mars` is captured as the value of `planet`.
```

Comment:
I wouldn't use the term "granular routing" - from a user POV this isn't a distinct 'granular routing' feature, it's just another kind of wildcard.  How about we change line 67 to something like:

-----

You can use wildcards to match 'patterns' of routes.  Spin supports two kinds of wildcard: _single-segment_ wildcards and _trailing_ wildcards.

A single-segment wildcard uses the syntax `:name`, where `name` is a name that identifies the wildcard.  Such a wildcard will match only a single segment of a path, and allows further matching on segments beyond it.  For example, `/users/:userid/edit` matches `/users/1/edit` and `/users/alice/edit`, but does not match `/users`, `/users/1`, or `/users/1/edit/cart`.

A _trailing wildcard_ uses the syntax `/...` and matches the given route and any route under it. (then continues as current para and example)

You can combine single-segment and trailing wildcards in the same route, e.g. `/users/:userid/cart/...`.

See Additional Request Information _(link)_ for how to access wildcard values from your component.

-----

(I've suggested changing your planets example to something that meshes a bit more with the running example of users and shops, and perhaps motivates the concept a bit more.  If you prefer to keep the planets then no worries.)

Reply:
Updating in the next push